src/mono/mono/mini/method-to-ir.c

```diff
@@ -4963,6 +4975,15 @@ inline_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsig,
 #define CHECK_UNVERIFIABLE(cfg) if (cfg->unverifiable) UNVERIFIED
 #define CHECK_TYPELOAD(klass) if (!(klass) || mono_class_has_failure (klass)) TYPE_LOAD_ERROR ((klass))
 
+#define CLEAR_TYPELOAD_EXCEPTION(cfg) if (cfg->exception_type == MONO_EXCEPTION_TYPE_LOAD) { clear_cfg_error (cfg); cfg->exception_type = MONO_EXCEPTION_NONE; }
+#define IF_TYPELOAD_ERROR(klass) if (!(klass) || mono_class_has_failure (klass))
```

Comment:
Would prefer something like
#define CLASS_HAS_FAILURE(klass) ((!klass) || mono_class_has_failure (klass))
, macros which contain control flow are kinda hard to read.